202312-quarto-reports/2023-12-quarto-reports.qmd

```diff
@@ -0,0 +1,278 @@
+
+    -   Or go to the URL: <https://cct-datascience.quarto.pub/quarto-workshop-example>
+
+-   [Source code for this example](https://github.com/diazrenata/squareone-quarto/blob/main/s3_biomass_as_quarto.qmd)
+
+    -   Or go to the URL: <https://github.com/diazrenata/squareone-quarto/blob/main/s3_biomass_as_quarto.qmd>
+
+::: notes
+Point out:
+
+-   Formatting
+-   In-line code, figures, tables
+-   Figure captions
+-   Citations
+-   Scholarly author tracking
+:::
+
+## Cool, but tricky. What are the benefits?
+
+::: incremental
+-   Reproducibility
+
+-   Dynamic updating
+
+-   Easy to share - from coauthors to journals to web audiences
+:::
+
+## What about RMarkdown?
+
+::: incremental
+-   Quarto is the "successor" to .Rmd
+-   .Rmd documents translate easily to Quarto
+-   *Quarto supports languages beyond R*
+-   *Quarto has a greatly expanded range of document types*
+:::
+
+## Comparison  {.smaller}
+
+| Feature                                                       | Quarto | Rmd | Markdown |
+|:-----------------------------------------|:---------|:---------|:---------|
+| Markup syntax (`**` for emphasis, etc)                        | x      | x   | x        |
+| Render to multiple document types                             | x      | x   | x        |
+| Interweave text and code                                      | x      | x   | x        |
+| Interweave text and *executed* R code                         | x      | x   |          |
+| Execute *non-R* code                                          | x      |     |          |
+| Expanded set of document types (dashboards, manuscripts, ...) | x      |     |          |
+
+## How do I create a Quarto document?
+
+Let's start from this R script: <https://github.com/cct-datascience/workshop-wednesdays/blob/2023-quarto-reports/20231206-quarto-reports/mouse_report.R>
+
+## .R to .Qmd 
+
+::: incremental
+
+1. Create a Quarto document
+1. Move code chunks over
+1. Add descriptive text
+1. Add references
+1. Add figure/table cross-references and captions
+
+:::
+
+## How do I turn an .Rmd into a Quarto document?
+
+Example report document available here: <https://github.com/cct-datascience/workshop-wednesdays/blob/2023-quarto-reports/20231206-quarto-reports/mouse_report_RMD.Rmd>
+
+
+## Technical changes from .Rmd to Quarto
+
+::: incremental
+**Most .Rmd documents will render as-is!**
+
+-   Document-level options moved to header
+
+-   `output` changed to `format`
+
+-   Chunk-level options moved from ```` ```{r ....} ```` to `#|` within the chunk
+
+-   `knitr::convert_chunk_header(<file_path>)` helps with conversion
+
+-   <https://quarto.org/docs/faq/rmarkdown.html>
+:::
+
+
+## .Rmd to .Qmd
+
+::: incremental
+
+1. Run `knitr::convert_chunk_header`
+1. Add additional Quarto features 
+
+:::
+
+## Sharing Quarto reports
+
+| Hosted online with a shareable URL | Static documents to share person-to-person |
```

Comment:
The table-y format of this is kind of confusing, I might just change this to be two bulleted lists on the slide?

Reply:
Done!